python/dataset.cpp

```diff
@@ -86,8 +86,11 @@ void init_dataset(py::module &m) {
   py::class_<DataConstProxy>(m, "DataConstProxy");
   py::class_<DataProxy, DataConstProxy> dataProxy(m, "DataProxy");
   dataProxy.def_property_readonly(
-      "data", py::cpp_function(&DataProxy::data, py::return_value_policy::move,
-                               py::keep_alive<0, 1>()));
+      "data", py::cpp_function(
+                  [](const DataProxy &self) -> py::object {
```

Comment:
Why is the trailing return type required here?

Reply:
It is actually not needed now, but the compiler was confused with the return type the nit was: 
```c++
if ()
  return py::cast();
else
  return py::none();
``` 
I suppose because the `py::none()` is not exact the `py::object' but just derived from it. But the ternary operator solves this.